assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * SurveyQuestionMultiSelect component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import keyBy from 'lodash/keyBy';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../Button';
+import Checkbox from '../Checkbox';
+import SurveyHeader from './SurveyHeader';
+import { TextField, Input } from '../../material-components';
+
+const MAXIMUM_CHARACTER_LIMIT = 100;
+
+/* eslint-disable camelcase */
+
+const SurveyQuestionMultiSelect = ( {
+	question,
+	choices,
+	answerQuestion,
+	dismissSurvey,
+	min_choices = 1,
+	max_choices,
+} ) => {
+	const mappedChoices = choices.map( ( { answer_ordinal, write_in } ) => {
+		const optionalKeys = write_in ? { answer_text: '' } : {};
```

Comment:
I think the point of this is to say that a "write-in" key is optional, but can you elaborate on this logic with a comment? It's just a bit unclear to me upon immediately reading this file, so a bit of explanation here would likely help future developers reading it 🙂